definitions/output/reports/cwv_tech_categories.js

```diff
@@ -8,43 +8,45 @@ publish('cwv_tech_categories', {
 /* {"dataform_trigger": "report_cwv_tech_complete", "name": "categories", "type": "dict"} */
 WITH pages AS (
   SELECT
-    root_page,
+    client,
+    root_page AS origin,
     technologies
   FROM ${ctx.ref('crawl', 'pages')}
   WHERE
-    date = '${pastMonth}' AND
-    client = 'mobile'
+    date = '${pastMonth}'
     ${constants.devRankFilter}
 ), categories AS (
   SELECT
+    client,
     category,
-    COUNT(DISTINCT root_page) AS origins
+    COUNT(DISTINCT origin) AS origins
   FROM pages,
-    UNNEST(technologies) AS t,
-    UNNEST(t.categories) AS category
-  GROUP BY category
+    UNNEST(technologies) AS tech,
+    UNNEST(tech.categories) AS category
+  GROUP BY
+    client,
+    category
 ), technologies AS (
   SELECT
+    client,
     category,
     technology,
-    COUNT(DISTINCT root_page) AS origins
-  FROM pages,
-    UNNEST(technologies) AS t,
-    UNNEST(t.categories) AS category
-  GROUP BY
-    category,
-    technology
+    origins
+  FROM ${ctx.ref('reports', 'cwv_tech_technologies')}
+  WHERE client = 'mobile'
 )
 
 SELECT
   category,
-  categories.origins,
-  ARRAY_AGG(technology IGNORE NULLS ORDER BY technologies.origins DESC) AS technologies
+  STRUCT(
+    COALESCE(MAX(IF(categories.client = 'desktop', categories.origins, 0))) AS desktop,
+    COALESCE(MAX(IF(categories.client = 'mobile', categories.origins, 0))) AS mobile
+  ) AS origins,
+  ARRAY_AGG(technologies.technology IGNORE NULLS ORDER BY technologies.origins DESC) AS technologies
 FROM categories
-JOIN technologies
+INNER JOIN technologies
 USING (category)
 GROUP BY
-  category,
-  categories.origins
-ORDER BY categories.origins DESC
+  category
+ORDER BY origins.mobile DESC
```

Comment:
Should we sub sort by desktop for tech that has the same number of mobile origins?

```suggestion
ORDER BY origins.mobile DESC, origins.desktop DESC, category
```

Would lead to more consistent results.

Reply:
Or maybe we remove sorting? We eventually rely on sorting it on the client.
And we do it `category ASC` there.

Reply:
Still think it's nice to sort in API too. But non-blocking so up to you.

Reply:
Agree, but that is to configure in the Firestore and function.
Let's do `category ASC` here too.